src/neo4j_graphrag/generation/prompts.py

```diff
@@ -196,3 +196,34 @@ def format(
         text: str = "",
     ) -> str:
         return super().format(text=text, schema=schema, examples=examples)
+
+
+class ChatSummaryTemplate(PromptTemplate):
+    DEFAULT_TEMPLATE = """
+Summarize the chat history:
+
+{chat_history}
+"""
+    EXPECTED_INPUTS = ["chat_history"]
+
+    def format(self, chat_history: list[dict[str, str]]) -> str:
+        message_list = [
+            ": ".join([f"{value}" for _, value in message.items()])
+            for message in chat_history
+        ]
+        history = "\n".join(message_list)
+        return super().format(chat_history=history)
+
+
+class ConversationTemplate(PromptTemplate):
+    DEFAULT_TEMPLATE = """
+Chat Summary: 
+{summary}
```

Comment:
Why is the chat summary required here? Isn't the chat history handling the past messages?

Reply:
Yes, but this template is for the query embedding. The chat history can potentially be large, so the idea is to first generate a summary and embed it together with the current query, instead of embedding the entire chat history.

Reply:
That's a great idea. I thought the ChatSummaryTemplate was used to summarized the chat history and embed it?

Reply:
The `ChatSummaryTemplate` is for summarizing the chat history, this summary is then used together with the user query for embedding, using the `ConversationTemplate`. See [graphrag.py](https://github.com/leila-messallem/neo4j-graphrag-python/blob/66b235872500fadbe727d6983695571146419e83/src/neo4j_graphrag/generation/graphrag.py#L150).

Reply:
Ok I see, this is only used in the Retriever. I got confused because `Template` for me refer to prompt template. They were introduced with the idea that users can define the template and the placeholders. But here we (developers) do both, so we could skip them and use simple Python string formatting. Up to you :)

Reply:
If it's confusing, I'll change it to a string.

Reply:
[bd34e1a](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/bd34e1ae4d882886d4da1433dbd967f3548b6357). Is it fine to keep those functions in the `prompts` module?

Reply:
If used in only one place, can be moved closer to where they are used. The convention for function names is to use `snake_case` :D

Reply:
Right 🤦‍♀️ 😄  [f5a9833](https://github.com/neo4j/neo4j-graphrag-python/pull/234/commits/f5a983373b3fcedecff9a575369286c038393873)